src/Lifecycle/Broker.php

```diff
@@ -18,8 +18,12 @@ public function __construct(
     ) {
     }
 
-    public function fire(Event $event): Event
+    public function fire(Event $event): ?Event
     {
+        if ($this->is_replaying) {
+            return null;
```

Comment:
Any reason that we couldn't just return the `Event` here to ensure `fire()` always returns consistently?

Reply:
I think we shouldn't return an event that wasn't fired.

Reply:
That was @DanielCoulbourne (we were pairing) ^^